### PR TITLE
Create test case consistency06 for SOA MNAME

### PR DESCRIPTION
**Pull request by matsduf**
There are test cases for the consistency of all SOA fields except SOA MNAME. This is a new test case under Consistency Test Plan that adds testing for that.

### CODE REVIEW

**Comment by mattias-p** [comment]
Why is this a new test case rather than an extra check in Consistency03? AFAICT it fits right into the pattern of equality checks performed in Consistency03.

**Comment by matsduf** [comment]
@mattias-p, can you please approve it? It is easier to keep track of which ones are done and which ones that are still open. I have also asked @johaniis to review the ones that are approved by at least one person.

@sandoche2k, can you please review it?

File: docs/specifications/tests/Consistency-TP/consistency06.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+* The domain name to be tested ("child zone")
+
+## Ordered description of steps to be taken to execute the test case
+
+ 1. Obtain the list of name server IPs for the *child zone* from [Method4] 
+    and [Method5].
+ 2. Create an SOA query for *child zone* apex and send it to all name 
+    server IPs.
+ 3. Retrieve the SOA RR from the responses from all name server IPs.
+ 4. If a name server does not respond, emit *[NO_RESPONSE]*.
```
Comment (mattias-p) [inline docs/specifications/tests/Consistency-TP/consistency06.md]: How could you retrieve the SOA RR if the name server didn't respond? Maybe put this step before the SOA RR retrieval step?

File: docs/specifications/tests/Consistency-TP/consistency06.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+ 5. If a name server responds but does not include a SOA record in 
+    the response, emit *[NO_RESPONSE_SOA_QUERY]*.
```
Comment (mattias-p) [inline docs/specifications/tests/Consistency-TP/consistency06.md]: How could you retrieve the SOA RR if was none? Maybe put this step before the SOA RR retrieval step?

File: docs/specifications/tests/Consistency-TP/consistency06.md (hunk 1 above)
Comment (matsduf) [reply docs/specifications/tests/Consistency-TP/consistency06.md]: Updated in new commit.

File: docs/specifications/tests/Consistency-TP/consistency06.md (hunk 2 above)
Comment (matsduf) [reply docs/specifications/tests/Consistency-TP/consistency06.md]: Updated in new commit.

File: docs/specifications/tests/Consistency-TP/consistency06.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+ 3. For each name server in *Name Server IP* do:
+
+    1. Send the query to name server.
+    2. If the name server does not respond with a DNS response, 
+       emit *[NO_RESPONSE]* for that name server and go to next server.
+    3. Retrieve the SOA RR from the response from the name server.
+    4. If the response does not include a SOA record in the ansser section
+       then emit *[NO_RESPONSE_SOA_QUERY]* for that server and go to next
+       server.
```
Comment (mattias-p) [inline docs/specifications/tests/Consistency-TP/consistency06.md]: How could you retrieve the SOA RR if there was none? Maybe put this step before the SOA RR retrieval step?

You said you'd made an update to address this, but maybe you forgot?

Comment (matsduf) [reply docs/specifications/tests/Consistency-TP/consistency06.md]: Please review again.

File: docs/specifications/tests/Consistency-TP/consistency06.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    2. If the name server does not respond with a DNS response, 
+       emit *[NO_RESPONSE]* for that name server and go to next server.
+    3. If the DNS response does not include a SOA record in the answer 
+       section then emit *[NO_RESPONSE_SOA_QUERY]* for that server and go 
+       to next server.
+    4. Retrieve the MNAME field from the SOA RR from the DNS response
+       and save that to compare it with the MNAME from the other name
+       servers.
+
+ 5. Compare the MNAME fields retreived from all name servers.
```
Comment (mattias-p) [inline docs/specifications/tests/Consistency-TP/consistency06.md]: The numbering is off.

Comment (matsduf) [reply docs/specifications/tests/Consistency-TP/consistency06.md]: Fixed.

**Pull request merged**